feat: add `getQuery` method to Firestore transactions for transaction…

kartikey321 opened the pull request:
## Summary

  - Add Transaction.getQuery() to enable query execution within Firestore transactions
  - capture transaction IDs from RunQueryResponse via new _QueryReader, mirroring _DocumentReader
  - and ship 10 new query-transaction tests covering mixed reads, read-only, limits, converters, and read-after-write enforcement

  ## Testing

  - dart test test/google_cloud_firestore/transaction_test.dart (Firestore emulator running)

Ehesp commented:
@kartikey321 Thanks for the contribution here - we're currently undergoing a large piece of work on the admin sdk (next branch), and we're currently working through Firestore.

There's a bunch of large changes which will be happening, one of which is the structure of the repo. We're also identifying API gaps to fill and this has come up as one of them (along with `getAggregateQuery`).

demolaf commented on packages/dart_firebase_admin/pubspec.yaml:
```diff
@@ -1,6 +1,6 @@
 name: dart_firebase_admin
 description: A Firebase Admin SDK implementation for Dart.
-version: 0.4.1
+version: 0.4.2
```
Since this would be included in the next release which has breaking changes there would be no need for this.

kartikey321 commented:
## Update: Rebased onto `next` branch 

Thanks for the feedback! I've addressed all your points:

### Changes Made:
1. **Rebased onto `next` branch** - Successfully adapted the implementation to work with the new `googleapis_firestore` package structure
2. **Version & Changelog** - Already reverted in previous commit (kept at 0.4.1, removed 0.4.2 entry)
3. **melos.yaml** - No longer an issue as it's removed in `next` branch
4.  Updated PR description to accurately reflect that this PR adds query transaction support, leveraging Firestore's existing pessimistic locking (not adding new locking mechanisms)

### Test Results:
-  All 20 transaction tests passing (including the 10 new query transaction tests)
-  Code compiles cleanly with no errors
-  Verified against Firestore emulator

### Code Changes for `next` branch compatibility:
- Migrated code from `dart_firebase_admin` to `googleapis_firestore` package
- Updated imports (`firestore1` → `firestore_v1`)
- Updated internal APIs (`_client` → `_firestoreClient`)
- Added `transactionOptions` parameter to `Query._toProto()` for lazy transaction initialization

The PR is now ready for re-review. Please let me know if you need any additional changes!

demolaf commented:
@kartikey321 the reason why this kept failing was some tests were running against prod which wasn't required, they've been fixed now, feel free to pull the next branch and hopefully the CI should pass now.

kartikey321 commented:
Thanks a lot @demolaf @Ehesp , in helping me merge this pr
